compiler/qsc/src/interpret/stateless.rs

```diff
@@ -72,28 +95,52 @@ impl Context {
         }
     }
 
+    #[must_use]
+    pub fn eval_context(&self) -> EvalContext {
```

Comment:
Thinking about the naming a bit more, I'm not sure that this API really communicates that you get a fresh eval context each time. It looks like a normal getter. Maybe something like `new_eval`?